website/docs/cdktf/concepts/resources.mdx

```diff
@@ -381,6 +381,236 @@ If you need to use the special [`self` object](/terraform/language/resources/pro
+                              ╵
+
+                              ─────────────────────────────────────────────────────────────────────────────
+
+                              Terraform has generated configuration and written it to
+                              generated_resources.tf. Please review the configuration and edit it as
+                              necessary before adding it to version control.
+
+                              Saved the plan to: plan
+
+                              To perform exactly these actions, run the following command to apply:
+                                  terraform apply "plan"
+ts-import-with-configuration  Import without configuration detected. Terraform has created configuration for it:
+                              # __generated__ by Terraform
+                              # Please review these resources and move them into your main configuration files.
+
+                              # __generated__ by Terraform from "best-bucket-in-the-world"
+                              resource "aws_s3_bucket" "bucket" {
+                                bucket              = "best-bucket-in-the-world"
+                                bucket_prefix       = null
+                                force_destroy       = null
+                                object_lock_enabled = false
+                                tags = {
+                                  foo = "bar"
+                                }
+                                tags_all = {
+                                  foo = "bar"
+                                }
+                              }
+
+
+                              CDKTF has translated the code to the following:
+
+                              import { Construct } from "constructs";
+                              /*
+                               * Provider bindings are generated by running `cdktf get`.
+                               * See https://cdk.tf/provider-generation for more details.
+                               */
+                              import { S3Bucket } from "./.gen/providers/aws/s3-bucket";
+                              class MyConvertedCode extends Construct {
+                                constructor(scope: Construct, name: string) {
+                                  super(scope, name);
+                                  new S3Bucket(this, "bucket", {
+                                    bucket: "best-bucket-in-the-world",
+                                    bucketPrefix: [null],
+                                    forceDestroy: [null],
+                                    objectLockEnabled: false,
+                                    tags: {
+                                      foo: "bar",
+                                    },
+                                    tagsAll: {
+                                      foo: "bar",
+                                    },
+                                  });
```

Comment:
This could be a follow-up issue as it would require changes to convert, but adding this code to the code base is not really enough – it would still require an `.importFrom()` call on the bucket to be able to apply after generating.